rust/src/encodings/plain.rs

```diff
@@ -129,13 +130,16 @@ impl<'a> PlainDecoder<'a> {
             DataType::Boolean => bit_util::ceil(self.length, 8),
             _ => get_primitive_byte_width(self.data_type)? * self.length,
         };
-        let range = Range {
-            start: self.position,
-            end: self.position + array_bytes,
-        };
+        let mut start = self.position;
+        let end = self.position + array_bytes;
+        let mut ranges: Vec<Range<usize>> = vec![];
+        while start < end {
+            ranges.push(start..min(start + 1 * 1024 * 1024, end));
+            start += 1 * 1024 * 1024;
+        }
 
-        let data = self.reader.get_range(range).await?;
-        let buf: Buffer = data.into();
+        let data = self.reader.get_ranges(&ranges).await?;
+        let buf: Buffer = data.concat().into();
```

Comment:
how expensive is this?

Reply:
Relative to the 4k random access, this is not that much. 

As mentioned above, we can make this cheaper via a native local filesystem implementation (`object_store` crate is not efficient in this regard).